### DISCUSSION TITLE
AWS Lambda Serverless for Application Load Balancer

### DISCUSSION THREAD

**Discussion by nkostrencic** [sqlpage/SQLPage#59] [Q&A]
Greetings,

I would love to use the serverless package with an ALB, but it seems when uploading the package it is meant to be used by an API gateway (when testing with API gateway events it works, when sending ALB events it returns an error). Is there something I'm missing or are there thoughts on implementing it?

Thanks for any replies!

**Comment by lovasoa** [comment]
Unfortunately ALB events are not supported by the library we use: https://github.com/hanabu/lambda-web

I am not an AWS expert, but if you help them add support for it in `lambda-web`, I'll be happy to then make the required changes on SQLPage's side.


This seems to be the file to change to add support:
https://github.com/hanabu/lambda-web/blob/main/src/request.rs

And the reference documentation for implementation
https://docs.aws.amazon.com/lambda/latest/dg/services-alb.html

**Reply by nkostrencic** [reply]
thanks for the suggestion! will look into it.
